program/src/instruction.rs

```diff
@@ -137,6 +137,40 @@ pub enum TokenWrapInstruction {
     /// 5. `[]` (Optional) Owner program. Required when metadata account is
     ///    owned by a third-party program.
     SyncMetadataToToken2022,
+
+    /// This instruction copies the metadata fields from an unwrapped mint to
+    /// its wrapped mint `Metaplex` metadata account.
+    ///
+    /// Supports (unwrapped to wrapped):
+    /// - Token-2022 to SPL-token
+    /// - SPL-token to SPL-token
+    ///
+    /// This instruction will create the `Metaplex` metadata account if it
+    /// doesn't exist, or update it if it does. The `wrapped_mint_authority`
+    /// PDA must be pre-funded with enough lamports to cover the rent for
+    /// the `Metaplex` metadata account's creation or updates, as it will
+    /// act as the payer for the `Metaplex` program CPI.
```

Comment:
Sadly, the metaplex metadata program [requires a payer](https://github.com/metaplex-foundation/mpl-token-metadata/blob/a7ee5e17ed60feaafeaa5582a4f46d9317c1b412/programs/token-metadata/program/src/processor/metadata/create_medatata_accounts_v3.rs#L30-L33) for the create instruction. Pre-funding the PDA is not an option, so this instruction uses the wrapped_mint_authority PDA as the signer and requires the user to transfer to that account.